File: firstspringdemo/src/main/java/firstspring/firstspringdemo/DTOs/LocationCreateDto.java
package firstspring.firstspringdemo.DTOs;

import java.util.UUID;

public class LocationCreateDto {
    private String name;
    private String address;
    private String city;
    private String state;
    private String country;
    private String postalCode;
    private UUID locationId; // Optional, if you want to set it manually

    public LocationCreateDto() {
    }

    public LocationCreateDto(String name, String address, String city, String state, String country, String postalCode, UUID locationId) {
        this.locationId = locationId;
        this.name = name;
        this.address = address;
        this.city = city;
        this.state = state;
        this.country = country;
        this.postalCode = postalCode;
    }

    public UUID getLocationId() {
        return locationId;
    }
    public void setLocationId(UUID locationId) {
        this.locationId = locationId;
    }
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }
}

File: firstspringdemo/flutter/lib/models/location_create_dto.dart
class LocationCreateDto {
  final String? name;
  final String? address;
  final String? city;
  final String? state;
  final String? country;
  final String? postalCode;
  final String? locationId;

  LocationCreateDto({
    this.name,
    this.address,
    this.city,
    this.state,
    this.country,
    this.postalCode,
    this.locationId,
  });

  factory LocationCreateDto.fromJson(Map<String, dynamic> json) {
    return LocationCreateDto(
      name: json['name'],
      address: json['address'],
      city: json['city'],
      state: json['state'],
      country: json['country'],
      postalCode: json['postalCode'],
      locationId: json['locationId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'address': address,
      'city': city,
      'state': state,
      'country': country,
      'postalCode': postalCode,
      'locationId': locationId,
    };
  }
}
